app/controllers/sessions_controller.rb

```diff
@@ -1,7 +1,7 @@
 class SessionsController < ApplicationController
   def new
     origin = { origin: session.delete(:return_to) }.to_query
-    redirect_to("/auth/google_oauth2?#{origin}")
+    redirect_to("/auth/#{ENV['OAUTH_PROVIDER_NAME']}?#{origin}")
```

Comment:
Same here, it would be good to call a config object rather than ENV vars deep inside controllers.